media/js/base/mozilla-fxa-iframe.js

```diff
@@ -0,0 +1,205 @@
+            _src = _src + '&email=' + encodeURIComponent(_config.userEmail);
+        }
+
+        // make sure _host URL doesn't have a trailing slash
+        _host = (_host[_host.length - 1] === '/') ? _host.substr(0, _host.length - 1) : _host;
+
+        // check user's Fx version to determine FxA iframe experience
+        if (Mozilla.Client.FirefoxMajorVersion >= 46) {
+            _src = _src.replace('context=iframe', 'context=fx_firstrun_v2');
+        }
+
+        // initialize GA event name
+        _gaEventName = _config.gaEventName || 'fxa';
+
+        // call iframe to life by setting src attribute
+        _$iframe.attr('src', _src);
+
+        // set a timeout to show FxA (error page, most likely) should the ping event fail
+        setTimeout(function() {
+            if (!_handshake) {
+                _showIframe(400);
+            }
+        }, 2500);
+    };
+
+    // resize and ensure visibility of iframe
+    var _showIframe = function(height) {
+        _$iframe.css('height', height + 'px').addClass('loaded');
+    };
+
+    var _sendGAEvent = function(type, extra) {
+        var data = {
+            'event': _gaEventName, // user-configurable, defaults to 'fxa'
+            'interaction': type
+        };
+
+        // merge additional properties if available
+        if (extra) {
+            $.extend(data, extra);
+        }
+
+        window.dataLayer.push(data);
+    };
+
+    // postMessage communication handlers
+    var _onMessageReceived = function(e) {
+        // make sure origin is as expected
+        if (e.origin !== _host) {
+            return;
+        }
+
+        var data = JSON.parse(e.data);
+
+        switch (data.command) {
+        // iframe is pinging host page
+        case 'ping':
+            _onPing(data);
+            break;
+        // iframe has loaded successfully
+        case 'loaded':
+            _onLoaded(data);
+            break;
+        // iframe has changed size due to content change
+        case 'resize':
+            _onResize(data);
+            break;
+        // user has signed up, but not yet verified email
+        case 'signup_must_verify':
+            _onSignupMustVerify(data);
+            break;
+        // user has returned to page after verifying email (may never happen)
+        case 'verification_complete':
+            _onVerificationComplete(data);
+            break;
+        // user has logged in (instead of signing up)
+        case 'login':
+            _onLogin();
+            break;
+        }
+    };
+
+    var _onPing = function(data) {
+        // tell iframe we are expecting it
+        if (!_config.testing) { // allow unit test to bypass auth
+            var fxaFrameTarget = _$iframe[0].contentWindow;
+            fxaFrameTarget.postMessage(data, _host);
+        }
+
+        // remember iframe has loaded
+        _handshake = true;
+
+        if (typeof _config.onPing === 'function') {
```

Comment:
Would it be worth adding a small utility `_isFunction()` method to avoid repeating the condition here and below? Something like

    function _isFunction(item) {
        return typeof item === 'function';
    }